runtime/src/main/scala/akka/grpc/internal/Marshallers.scala

```diff
@@ -28,11 +28,27 @@ final class Marshaller[T <: scalapb.GeneratedMessage](u: ProtobufSerializer[T])
       baos.write(buffer, 0, bytesRead)
       bytesRead = stream.read(buffer)
     }
-    u.deserialize(akka.util.ByteString(baos.toByteArray))
+    protobufSerializer.deserialize(akka.util.ByteString(baos.toByteArray))
   }
 
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalStableApi
+final class Marshaller[T <: scalapb.GeneratedMessage](protobufSerializer: ProtobufSerializer[T])
+    extends BaseMarshaller[T](protobufSerializer) {
   override def stream(value: T): InputStream =
     new ByteArrayInputStream(value.toByteArray) with KnownLength
```

Comment:
I was tempted to use:

```
    new ByteArrayInputStream(protobufSerializer.serialize(value).toArray) with KnownLength
```

and move this method to the base class but I'd be causing unnecessary allocations (see #1180)